control-plane/rest/service/src/v0/pools.rs

```diff
@@ -1,19 +1,18 @@
 use super::*;
 use common_lib::types::v0::message_bus::{DestroyPool, Filter};
-use mbus_api::{
-    message_bus::v0::{BusError, MessageBus, MessageBusTrait},
-    ReplyErrorKind, ResourceKind,
-};
+use grpc::pool::traits::PoolOperations;
+use mbus_api::{message_bus::v0::BusError, ReplyErrorKind, ResourceKind};
 
 async fn destroy_pool(filter: Filter) -> Result<(), RestError<RestJsonError>> {
+    let client = CORE_CLIENT.get().unwrap().pool();
     let destroy = match filter.clone() {
         Filter::NodePool(node_id, pool_id) => DestroyPool {
             node: node_id,
             id: pool_id,
         },
         Filter::Pool(pool_id) => {
-            let node_id = match MessageBus::get_pool(filter).await {
-                Ok(pool) => pool.node(),
+            let node_id = match client.get(filter, None).await {
+                Ok(pools) => pools.into_inner()[0].clone().node(),
```

Comment:
nit: I would use get(0) here as the slice will panic on an empty vec.

Reply:
But then we have to handle the unwrap again right?

Reply:
```
match pool.into_inner().get(0) {
            None => Err(ReplyError {
                kind: ReplyErrorKind::NotFound,
                resource: ResourceKind::Pool,
                source: "Requested pool was not found".to_string(),
                extra: format!("Pool id : {}", pool_id),
            }
            .into()),
            Some(pool) => Ok(pool.clone().into()),
        }
```
Maybe something like this?

Reply:
Yeah, you can put that on a function, eg: fn pool(i: Option<&Pool>) -> Result<Pool, Error>, and then just pool(xxx)? on all the places where we need to do it

Reply:
I wonder if we should have a get that returns a single pool/replica.

Reply:
even if we create a get that returns single, at some point be it at service or client we have to extract from a vector as grpc req to mayastor always returns a vector.